azure-pipelines.yml

```diff
@@ -0,0 +1,71 @@
+# Node.js with Angular
+# Build a Node.js project that uses Angular.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
+variables:
+  appDir: 'fakebook/'
+
+trigger:
+  branches:
+   include: 
+     - main
+  paths:
+    include:
+      - az-pipeline-angular.yml
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+stages:
+  - stage: build
+    jobs:
+    - job: build
+      variables:
+        imageName: 'fakebookng'
+      
+      steps:
+      - task: Docker@2
+        displayName: Build Docker Image
+        inputs:
+          repository: $(imageName)
+          command: 'build'
+          Dockerfile: '$(appdir)Dockerfile'
+  
+  - stage: SonarCloud
+    jobs:
+    - job: build
+      steps:
+      - task: NodeTool@0
+        inputs:
+          versionSpec: '10.x'
```

Comment:
node 10 is old, use node 15 since it has npm 7 and we're using lockfileversion 2.
on that note, the dockerfile should also use node 15, so fix that too.
add comments both here and in the dockerfile indicating why.

Reply:
using node 15.x in the pipeline, will update and do a PR for docker